### PR TITLE
Fixed issue with axis chart grouping repeating data

### PR DESCRIPTION
**Pull request by jez500**
See DVFD9-53, fixes incorrect data when split charts

### CODE REVIEW

**Comment by jez500** [comment]
@WilsonWu2100 this should do the trick!

File: src/Plugin/Visualisation/Style/VisualisationStyleBase.php [hunk 1]
```diff
@@ -334,20 +334,19 @@ protected function getSourceFieldOptions() {
    *
    * @param string $field_id
    *   The field ID.
+   * @param array $records
+   *   The set of records that we should get the values from.
    *
    * @return array
    *   The source field values.
    */
-  protected function getSourceFieldValues($field_id) {
```
Comment (jez500) [inline src/Plugin/Visualisation/Style/VisualisationStyleBase.php:341]: @WilsonWu2100 this was the culprit, not that array_merge. All records were getting added to `$values`, now it gets passed the correct set of records to act on

Comment (WilsonWu2100) [reply src/Plugin/Visualisation/Style/VisualisationStyleBase.php:341]: @jez500 The fix looks pretty good and it seems the correct way to solve the problem, I am a bit scary of touching the base code, which might break something else. But anyway, thanks for your help.

**Review by WilsonWu2100 — COMMENTED** [review]
@jez500 I've tested, it's working. Thanks.

**Comment by jez500** [comment]
Hey @WilsonWu2100 I have added the fix for the download buttons now too, can you please test

File: src/Plugin/Visualisation/Style/AxisChart.php [hunk 2]
```diff
@@ -742,30 +742,39 @@ public function build() {
     $build = [];
 
     foreach ($this->getSourceRecords() as $group_key => $group_records) {
-      $build[$group_key]['chart'] = [
+      $group_id = strtolower(Html::cleanCssIdentifier($group_key));
+
+      $build[$group_id] = [
```
Comment (jez500) [inline src/Plugin/Visualisation/Style/AxisChart.php:747]: @WilsonWu2100 this was the main issue for the download buttons, each split chart didn't have its own wrapper, meaning jQuery just grabbed the first chart

Comment (WilsonWu2100) [reply src/Plugin/Visualisation/Style/AxisChart.php:747]: @jez500 I've tested this, it's working perfectly. Thank you.

**Pull request merged**